pinot-servicemanager/build.gradle.kts

```diff
@@ -1,9 +1,41 @@
 plugins {
+  `java-library`
   id("org.hypertrace.docker-publish-plugin")
 }
 
 hypertraceDocker {
   defaultImage {
     imageName.set("pinot-servicemanager")
+    tasks.named(buildTaskName) {
+      dependsOn("copyPlugins")
+    }
   }
 }
+
+configure<JavaPluginExtension> {
+  sourceCompatibility = JavaVersion.VERSION_11
+  targetCompatibility = JavaVersion.VERSION_11
+}
+
+val plugins by configurations.creating
+
+dependencies {
+  compileOnly("org.apache.pinot:pinot-spi:0.5.0")
+  compileOnly("org.apache.pinot:pinot-avro-base:0.5.0")
+  compileOnly("org.apache.kafka:kafka-streams:5.5.1-ccs")
+  compileOnly("org.apache.kafka:kafka-clients:5.5.1-ccs")
+  implementation("org.hypertrace.core.kafkastreams.framework:kafka-streams-serdes:0.1.11") {
+    // disable the transitive dependencies and use them from pinot itself.
```

Comment:
This is always a risky move, as we're now forced to ensure that the dependencies of this module and those existing in pinot are in sync, when either one can move.

Reply:
Yes, we understand your concern, and we internally had this discussion before we proceed with this implementation. 

This Serde implementation depends on the following interfaces.
1. Serde from kafka stream
2. Serializer and Deserializer from kafka client
3. StreamMessageDecoder, RecordExtractor from pinot

As the above interfaces are public and stable interfaces, the risk involved incompatibility is very low and this Serde is used only in HT quickstart mode. Secondly, we are planning to contribute - GenericAvroMessageDecoder - to pinot-community for quickstart mode.

cc: @laxmanchekka